src/s390x/zipl.rs

```diff
@@ -28,109 +27,54 @@ use walkdir::WalkDir;
 /// # Arguments
 /// * `boot` - path to boot partition mountpoint, i.e. smth like /boot
 /// * `disk` - smth like /dev/dasda
+/// * `firstboot_kargs` - enables ignition firstboot and adds provided kargs
 pub fn install_bootloader<P: AsRef<Path>>(
     boot: P,
-    disk: &str,
-    extra_kargs: Option<&str>,
+    disk: Option<&str>,
+    firstboot_kargs: Option<&String>,
 ) -> Result<()> {
     eprintln!("Installing bootloader");
-
-    let bls_config_path = get_bls_config_path(&boot)?;
-    let kernel_path = get_kernel_path(&boot)?;
-    let initramfs_path = get_initramfs_path(&boot)?;
-
-    let kargs = format!(
-        "{} ignition.firstboot {}",
-        get_boot_kargs(bls_config_path)?,
-        extra_kargs.unwrap_or("")
+    let boot = boot.as_ref();
+    let conffile = "/tmp/zipl.conf";
```

Comment:
We should use the `tempfile` API instead here.

Reply:
we reboot machine right after `zipl`, so `/tmp` is clean again on the next boot. do we really need to use `tempfile` here?